File: backend/src/main/java/br/gov/ac/tce/licon/entities/Contrato.java
package br.gov.ac.tce.licon.entities;

import lombok.*;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.validation.constraints.NotNull;
import java.time.LocalDate;


@Data
@EqualsAndHashCode(callSuper = true)
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Contrato extends AbstractAuditableEntity {

	@NotNull
	@Column(name = "TITULO")
	private String titulo;

	@NotNull
	@Column(name = "DESCRICAO")
	private String descricao;

	@NotNull
	@Column(name = "PERMITE_ADITIVO")
	private Boolean permiteAditivo;

	@NotNull
	@Column(name = "DATA_VENCIMENTO")
	private LocalDate dataVencimento;

}

File: frontend/src/domains/Contrato.js
import { observable } from 'mobx';
import DomainBase from './DomainBase';

class Contrato extends DomainBase {
  @observable id;
  @observable titulo;
  @observable descricao;
  @observable permiteAditivo;
  @observable dataVencimento;

  static getDomainAttributes() {
    return ['id', 'titulo', 'descricao', 'permiteAditivo', 'dataVencimento'];
  }
}

export default Contrato;
